### PR TITLE
[bug 1168475] Fix allow_blank on char fields

### PR DESCRIPTION
**Pull request by willkg**
* fix allow_blank=True on char fields that should allow blanks
* clean up some other code in what should be cosmetic ways

r?

### CODE REVIEW

**Comment by willkg** [comment]
This needs a test to verify the behavior. I'll write that tomorrow. Does this look right-ish, though?

File: fjord/feedback/models.py [hunk 1]
```diff
@@ -634,41 +634,41 @@ class PostResponseSerializer(serializers.Serializer):
     url = serializers.CharField(max_length=200, allow_blank=True, default=u'')
     description = serializers.CharField(required=True)
 
-    category = serializers.CharField(max_length=50, required=False,
-                                     default=u'')
+    category = serializers.CharField(
+        max_length=50, allow_null=False, allow_blank=True, default=u'')
```
Comment (rlr) [inline fjord/feedback/models.py]: ah, you did what I was looking at. :+1:

**Comment by rlr** [comment]
lgtm r+

**Comment by willkg** [comment]
^^^ That nixes the `allow_null=False` because it's the default anyhow. It also adds a test for sending a blank category. Previously that would kick up an HTTP 400. Now we're ok with it.

**Comment by rlr** [comment]
r+ if travis likes it!

**Comment by willkg** [comment]
Thank you!

**Pull request merged**